src/mail/easey-content-template.service.ts

```diff
@@ -1,44 +1,149 @@
 @Injectable()
 export class EaseyContentTemplateService {
+  private handlebars = Handlebars.create();
+  private partialsRegistered = false;
+
   constructor(
     private readonly entityManager: EntityManager,
-    private readonly nodemailerService: NodemailerService,
     private readonly configService: ConfigService,
     private readonly httpService: HttpService,
     private readonly logger: Logger,
-  ) {}
+  ) {
+    // Register custom helpers for the isolated handlebars instance
+    this.registerHelpers();
+  }
+
+  private registerHelpers() {
+    // Register the notEquals helper needed by templates
+    this.handlebars.registerHelper('notEquals', function(a, b) {
+      return a !== b;
+    });
+
+    // Register the eq helper
+    this.handlebars.registerHelper('eq', function(a, b) {
+      return a === b;
+    });
+
+    // Register the equals helper (alternative to eq)
+    this.handlebars.registerHelper('equals', function(a, b) {
+      return a === b;
+    });
+
+    // Register the not_eq helper (alternative to notEquals)
+    this.handlebars.registerHelper('not_eq', function(a, b) {
+      return a !== b;
+    });
+
+    // Register the and helper for logical AND operations
+    this.handlebars.registerHelper('and', function() {
+      return Array.prototype.every.call(arguments, Boolean);
+    });
+
+    // Register the or helper for logical OR operations
+    this.handlebars.registerHelper('or', function() {
+      return Array.prototype.slice.call(arguments, 0, -1).some(Boolean);
+    });
+
+    // Register isObject helper to check if value is an object
+    this.handlebars.registerHelper('isObject', function(value) {
+      return typeof value === 'object' && value !== null;
+    });
+  }
+
+  private async registerPartials() {
+    // Register all configured partials
+    for (const [templateType, config] of Object.entries(EMAIL_TEMPLATE_PARTIALS)) {
+      this.logger.debug(`Registering partials for ${templateType}`);
+      
+      for (const partialName of config.partials) {
+        try {
+          const partialPath = `${config.basePath}/${partialName}.hbs`;
+          const partialContent = await this.getTemplateContent(partialPath);
+          this.handlebars.registerPartial(partialName, partialContent);
+          this.logger.debug(`Registered partial: ${partialName}`);
+        } catch (error) {
+          this.logger.error(`Failed to register partial ${partialName} from ${config.basePath}`, error);
+        }
+      }
+    }
+  }
 
   returnManager() {
     return this.entityManager;
   }
 
-  async getAndFormatTemplate(templateUrl, context): Promise<string> {
+  // Handlebars template methods
+  async getTemplateContent(templateLocation: string): Promise<string> {
+    const contentUri = this.configService.get<string>('app.contentUri');
+    try {
+      // Handle trailing/leading slashes properly
+      const url = new URL(templateLocation, contentUri).toString();
```

Comment:
This isn't resolving the path correctly for me when serving the content from a sub-path without a trailing slash, for example "http://localhost:80/api". From the [docs](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL):
> Relative references to the parent and current directory are resolved relative to the current directory of the base URL, which includes path segments **up until the last forward-slash, but not any after.**

I can fix this by adding a trailing slash in the environment variable, so the default could be updated (it's currently set as "https://api.epa.gov/easey/dev/content-mgmt"), but I think it would be better to add it here:
```suggestion
      const url = new URL(templateLocation, `${contentUri}/`).toString();
```
Or use a function that ensures a single trailing slash.

Reply:
Yep! I have updated and pushed this change.